src/analyses/variable-sensitivity/abstract_pointer_object.h

```diff
@@ -2,84 +2,52 @@
 
  Module: analyses variable-sensitivity
 
- Author: Thomas Kiley, thomas.kiley@diffblue.com
+ Author: Jez Higgins, jez@jezuk.co.uk
 
 \*******************************************************************/
 
 /// \file
 /// The base of all pointer abstractions
-#ifndef CPROVER_ANALYSES_VARIABLE_SENSITIVITY_POINTER_ABSTRACT_OBJECT_H
-#define CPROVER_ANALYSES_VARIABLE_SENSITIVITY_POINTER_ABSTRACT_OBJECT_H
-
-#include <stack>
+#ifndef CPROVER_ANALYSES_VARIABLE_SENSITIVITY_ABSTRACT_POINTER_OBJECT_H
+#define CPROVER_ANALYSES_VARIABLE_SENSITIVITY_ABSTRACT_POINTER_OBJECT_H
 
 #include <analyses/variable-sensitivity/abstract_object.h>
 
-class typet;
-class constant_exprt;
-class abstract_environmentt;
+class abstract_pointer_tag
+{
+};
 
-class pointer_abstract_objectt : public abstract_objectt
+class abstract_pointer_objectt : public abstract_objectt,
+                                 public abstract_pointer_tag
 {
 public:
   /// \param type: the type the abstract_object is representing
-  explicit pointer_abstract_objectt(const typet &type);
+  explicit abstract_pointer_objectt(const typet &type);
 
   /// Start the abstract object at either top or bottom or neither
   /// Asserts if both top and bottom are true
   ///
   /// \param type: the type the abstract_object is representing
   /// \param top: is the abstract_object starting as top
   /// \param bottom: is the abstract_object starting as bottom
-  pointer_abstract_objectt(const typet &type, bool top, bool bottom);
-
-  /// Interface for transforms
-  ///
-  /// \param expr: the expression to evaluate and find the result of it.
-  ///              This will be the symbol referred to be op0()
-  /// \param operands: an abstract_object (pointer) that represent
-  ///                  the possible values of each operand
-  /// \param environment: the abstract environment in which the
-  ///                     expression is being evaluated
-  /// \param ns: the current variable namespace
-  ///
-  /// \return Returns the abstract_object representing the result of
-  ///         this expression to the maximum precision available.
-  ///
-  /// To try and resolve different expressions with the maximum level
-  /// of precision available.
-  abstract_object_pointert expression_transform(
-    const exprt &expr,
-    const std::vector<abstract_object_pointert> &operands,
-    const abstract_environmentt &environment,
-    const namespacet &ns) const override;
+  abstract_pointer_objectt(const typet &type, bool top, bool bottom);
 
   /// \param expr: the expression to use as the starting pointer for
   ///              an abstract object
   /// \param environment: the environment in which the pointer is being
   ///                     created
   /// \param ns: the current namespace
-  explicit pointer_abstract_objectt(
+  explicit abstract_pointer_objectt(
     const exprt &expr,
     const abstract_environmentt &environment,
     const namespacet &ns);
 
-  /**
```

Comment:
If the function stays then it would be good if the doxygen stays / gets updated.